src/network/connectionthreads.cpp

```diff
@@ -1208,6 +1208,10 @@ SharedBuffer<u8> ConnectionReceiveThread::handlePacketType_Control(Channel *chan
 			m_connection->SetPeerID(peer_id_new);
 		}
 
+		// set non legacy mode locally
+		dynamic_cast<UDPPeer *>(peer)->setNonLegacyPeer();
+
+		// request the same from the remote side
```

Comment:
request?
so if the remote side does not support this, wouldn't we need to omit the `setNonLegacyPeer()` call?

Reply:
The sending sliding window for packets is independent on the other side also have a sliding window. It simply means that that many packets can be outstanding.

... I think ...  There is *soo* much code around this.

Reply:
We can also just remove the whole legacy peer nonsense going forward.

Reply:
Yes, that would be much preferred.

Reply:
As far as I can see this was wrong from the beginning (looked through the git history). So yes, I think I think it does matter, but I'll admit I'm running out of my depth there. We had removed a bunch of the old protocol stuff a while ago, so perhaps this was left over.

Lemme amend this with a complete removal.

Reply:
Turned out to be OK: #7551

Reply:
It would still be useful to merge this first to make it easy to backport the fix (also so it won't be forgotten).

Reply:
Agreed. Will do that now.